pyfqmr/Simplify.pyx

```diff
@@ -239,3 +244,25 @@ cdef vector[vector[int]] setFacesNogil(int[:,:] faces, vector[vector[int]] vecto
             triangle.push_back(faces[i,j])
         vector_faces.push_back(triangle)
     return vector_faces
+
+
+# For faster type checking
+def is_integer_array(arr):
+    # Check if the dtype is integer before calling Cython function
+    if np.issubdtype(arr.dtype, np.integer):
+        return True
+    return is_integer_array_cython(arr)
```

Comment:
Surely we don't really want to support faces as floats, even if they do happen to all be whole numbers?

Reply:
I thought that given the flexibility of python, numpy based meshprocessing software wouldn't really care if the dtype of the faces array is float or int, but it would make sense to have them always as int. 
So you would rather prefer throwing an error if faces type is not of integer sub dtype ? That would be easier and faster for sure, and may help safeguard some mistakes. But it is true that the user must be sure that the arrays he is passing make some geometric sense and represent a mesh, otherwise the behaviour may be unpredictable (and produce segfaults).